elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -337,8 +338,9 @@ public String getTableVersion(Table table, Transaction transaction) {
      */
     private NativeQuery toSQL(Query query, SQLDialect sqlDialect) {
         SQLReferenceTable queryReferenceTable = new DynamicSQLReferenceTable(referenceTable, query);
+        TableContext context = TableContext.tableContextBuilder().queryable(query).build();
 
-        QueryTranslator translator = new QueryTranslator(queryReferenceTable, sqlDialect);
+        QueryTranslator translator = new QueryTranslator(queryReferenceTable, sqlDialect, context);
```

Comment:
Maybe QueryTranslator should just construct the TableContext directly instead of it being passed in